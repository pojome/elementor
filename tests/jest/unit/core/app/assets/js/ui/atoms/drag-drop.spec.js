import { render, fireEvent } from '@testing-library/react';
import DragDrop from 'elementor-app/ui/atoms/drag-drop';

describe( '<DragDrop />', () => {
	it( 'should have only base classes when there is no interaction with the drag area', () => {
		// Arrange + Act
		const { container } = render( <DragDrop /> );

		// Assert
		expect( container.firstChild ).toHaveClass( 'e-app-drag-drop' );
		expect( container.firstChild ).not.toHaveClass( 'e-app-drag-drop--drag-over' );
	} );

	it( 'should have drag over class and call drag over prop when dragging over', () => {
		// Arrange
		const mockDragOver = jest.fn();
		const { container } = render( <DragDrop onDragOver={mockDragOver} /> );

		// Act
		fireEvent.dragOver( container.firstChild );

		// Assert
		expect( container.firstChild ).toHaveClass( 'e-app-drag-drop' );
		expect( container.firstChild ).toHaveClass( 'e-app-drag-drop--drag-over' );
		expect( mockDragOver ).toHaveBeenCalled();
	} );

	it( 'should not add drag over class if the component in loading state', () => {
		// Arrange
		const { container } = render( <DragDrop isLoading={true} /> );

		// Act
		fireEvent.dragOver( container.firstChild );

		// Assert
		expect( container.firstChild ).toHaveClass( 'e-app-drag-drop' );
		expect( container.firstChild ).not.toHaveClass( 'e-app-drag-drop--drag-over' );
	} );

	it( 'should call onDrop prop when drop event fired', () => {
		// Arrange
		const mockDrop = jest.fn();
		const { container } = render( <DragDrop onDrop={mockDrop} /> );

		// Act
		fireEvent.drop( container.firstChild );

		// Assert
		expect( mockDrop ).toHaveBeenCalled();
	} );

	it( 'should call onDragLeave prop when dragLeave event fired', () => {
		// Arrange
		const mockDragLeave = jest.fn();
		const { container } = render( <DragDrop onDragLeave={mockDragLeave} /> );

		// Act
		fireEvent.dragLeave( container.firstChild );

		// Assert
		expect( mockDragLeave ).toHaveBeenCalled();
	} );

	it( 'should match the snapshot', () => {
		// Arrange
		const { container } = render(
			<DragDrop className="test" isLoading={false}>
				<div>
					This is a test
				</div>
			</DragDrop>
		);

		// Act + Assert
		expect( container ).toMatchSnapshot();
	} );
} );
