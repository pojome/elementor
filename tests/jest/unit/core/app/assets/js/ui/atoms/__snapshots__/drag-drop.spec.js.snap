// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DragDrop /> should match the snapshot 1`] = `
<div>
  <div
    class="e-app-drag-drop test"
  >
    <div>
      This is a test
    </div>
  </div>
</div>
`;
